<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Christina B Azodi" />


<title>WEEO: QTLExperiment and multistateQTL manuscript</title>

<script src="site_libs/header-attrs-2.15/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>










<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">WEEO_2021_multi-QTL-benchmark</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2021_meta-eQTL">
    <span class="fa fa-gitlab"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">WEEO: QTLExperiment and multistateQTL
manuscript</h1>
<h4 class="author">Christina B Azodi</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 1
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>WEEO_2021_meta-eQTL/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has staged changes. To know which version of the
R Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgitlabsvieduaubiocellgenprivateWEEO2022multistateQTLtree83405215772a39a777bd477308feb9ba4b0b9f14targetblank8340521a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/tree/83405215772a39a777bd477308feb9ba4b0b9f14" target="_blank">8340521</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgitlabsvieduaubiocellgenprivateWEEO2022multistateQTLtree83405215772a39a777bd477308feb9ba4b0b9f14targetblank8340521a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/tree/83405215772a39a777bd477308feb9ba4b0b9f14" target="_blank">8340521</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    Manuscript-Figures_files/
    Ignored:    data/Adipose_Subcutaneous.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Adipose_Subcutaneous.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Adipose_Visceral_Omentum.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Adrenal_Gland.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Artery_Aorta.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Artery_Coronary.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Artery_Tibial.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Artery_Tibial.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Amygdala.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Anterior_cingulate_cortex_BA24.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Caudate_basal_ganglia.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Cerebellar_Hemisphere.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Cerebellum.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Cortex.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Frontal_Cortex_BA9.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Hippocampus.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Hypothalamus.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Nucleus_accumbens_basal_ganglia.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Putamen_basal_ganglia.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Spinal_cord_cervical_c-1.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Substantia_nigra.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Breast_Mammary_Tissue.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Cells_Cultured_fibroblasts.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Cells_EBV-transformed_lymphocytes.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Colon_Sigmoid.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Colon_Transverse.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Esophagus_Gastroesophageal_Junction.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Esophagus_Mucosa.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Esophagus_Mucosa.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Esophagus_Muscularis.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/GTEx_Analysis_v8_eQTL_all_associations/
    Ignored:    data/Heart_Atrial_Appendage.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Heart_Left_Ventricle.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Kidney_Cortex.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Liver.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Lung.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Lung.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Lung.v8.signif_variant_gene_pairs_chr1.txt
    Ignored:    data/Minor_Salivary_Gland.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Muscle_Skeletal.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Muscle_Skeletal.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Nerve_Tibial.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Nerve_Tibial.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Ovary.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Pancreas.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Pituitary.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Prostate.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Skin_Not_Sun_Exposed_Suprapubic.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Skin_Not_Sun_Exposed_Suprapubic.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Skin_Sun_Exposed_Lower_leg.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Skin_Sun_Exposed_Lower_leg.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Small_Intestine_Terminal_Ileum.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Spleen.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Spleen.v8.signif_variant_gene_pairs_chr1.txt
    Ignored:    data/Stomach.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Testis.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Thyroid.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Thyroid.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Thyroid.v8.signif_variant_gene_pairs_chr1.txt
    Ignored:    data/Uterus.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Vagina.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Whole_Blood.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Whole_Blood.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Whole_Blood.v8.signif_variant_gene_pairs_chr1.txt
    Ignored:    output/00_Figures/
    Ignored:    output/01_simulations/
    Ignored:    output/gtex10_chr1_QTLE.rds
    Ignored:    output/gtex10_chr1_QTLE_filt.rds
    Ignored:    output/gtex_estimated_params.rds
    Ignored:    references/chr2.filtered.log
    Ignored:    references/chr2.filtered.vcf
    Ignored:    references/chr2.filtered2.log
    Ignored:    references/chr2.filtered2.vcf
    Ignored:    references/chr2.filtered3.log
    Ignored:    references/chr2.filtered3.vcf
    Ignored:    references/chr2.genes.gff3
    Ignored:    references/chr2.vcf.gz

Untracked files:
    Untracked:  .gitignore
    Untracked:  WEEO_2021_meta-eQTL.Rproj
    Untracked:  analysis/02_simulation-analysis.Rmd
    Untracked:  analysis/2021-08-03_pbmc8k-simulations.Rmd
    Untracked:  code/preprocess_gtex.sh
    Untracked:  site_libs/
    Untracked:  workflow/

Unstaged changes:
    Deleted:    Snakefile
    Modified:   analysis/01_GTEx-analysis.Rmd
    Modified:   analysis/2021-07-30_pbmc8k-reference-processing.Rmd
    Modified:   analysis/index.Rmd

Staged changes:
    New:        analysis/01_GTEx-analysis.Rmd
    Modified:   analysis/index.Rmd
    New:        code/preprocess_gtex_alltests.sh
    Deleted:    code/sce_plots.R
    Deleted:    code/sim_celltypes.R
    Deleted:    code/sim_celltypes.Rmd
    Deleted:    envs/myenv.yaml
    Deleted:    resources/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>public/index.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/blob/eb6e33cf1f8ea7a52c1b26d48024a8ae193329ad/public/index.html" target="_blank">eb6e33c</a>
</td>
<td>
cazodi
</td>
<td>
2021-07-30
</td>
<td>
estimate pbmc8k splatPop params
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/blob/0032c48d3f4d3c91570db14155cbea138797f79c/public/index.html" target="_blank">0032c48</a>
</td>
<td>
cazodi
</td>
<td>
2021-07-30
</td>
<td>
estimate pbmc8k splatPop params
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/blob/a23584f023894660c13b18f282ddf1769ea97f56/analysis/index.Rmd" target="_blank">a23584f</a>
</td>
<td>
cazodi
</td>
<td>
2021-07-30
</td>
<td>
estimate pbmc8k splatPop params
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/blob/a23584f023894660c13b18f282ddf1769ea97f56/public/index.html" target="_blank">a23584f</a>
</td>
<td>
cazodi
</td>
<td>
2021-07-30
</td>
<td>
estimate pbmc8k splatPop params
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/blob/b200a1f3bdff39cbc3cf6743f030d6960f7ce619/analysis/index.Rmd" target="_blank">b200a1f</a>
</td>
<td>
cazodi
</td>
<td>
2020-03-18
</td>
<td>
removing accidental commits by christina…
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/blob/9057b09a6b2e7812472c30ec2ac2207074033e71/analysis/index.Rmd" target="_blank">9057b09</a>
</td>
<td>
cazodi
</td>
<td>
2020-03-18
</td>
<td>
Revert "initial repo change"
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/blob/0c6d3c6db6441404af9b70cce96dce19ed0d5333/analysis/index.Rmd" target="_blank">0c6d3c6</a>
</td>
<td>
cazodi
</td>
<td>
2020-03-18
</td>
<td>
initial repo change
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/blob/c57ac08311a07a29bbe2db43cf50d0f67540a4c0/analysis/index.Rmd" target="_blank">c57ac08</a>
</td>
<td>
Christina Azodi
</td>
<td>
2019-11-26
</td>
<td>
First init
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/blob/b1f309cd921c5423049b786ae95d2540a891f0c5/analysis/index.Rmd" target="_blank">b1f309c</a>
</td>
<td>
Davis McCarthy
</td>
<td>
2019-02-06
</td>
<td>
Updating template with improvements.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/blob/d73fd375ae2e4e63d90b25ec4cc76044396f5b4a/analysis/index.Rmd" target="_blank">d73fd37</a>
</td>
<td>
Davis McCarthy
</td>
<td>
2019-01-10
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="project-overview" class="section level2">
<h2>Project overview</h2>
<ol style="list-style-type: decimal">
<li>QTLExperiment; a container for storing and manipulating QTL summary
statistics and associated metadata; and</li>
<li>multistateQTL; a toolkit for simulating, analyzing, and visualizing
multi-state QTL summary statistics.</li>
</ol>
</div>
<div id="empirical-data-pre-processing" class="section level2">
<h2>1. Empirical Data Pre-processing</h2>
<p>The GTEx v8 data was downloaded from <a
href="https://console.cloud.google.com/storage/browser/gtex-resources/GTEx_Analysis_v8_QTLs/GTEx_Analysis_v8_eQTL_all_associations?pageState=(%22StorageObjectListTable%22%3A(%22f%22%3A%22%255B%255D%22))&amp;prefix&amp;forceOnObjectsSortingFiltering=false&amp;cloudshell=true">Google
Cloud</a>) and then filtered to keep only associations tested on
chromosome 1. This processing is described in
<code>code/preprocess_gtex_alltests.sh</code>.</p>
</div>
<div id="gtex-examples" class="section level2">
<h2>2. GTEx examples</h2>
<p>See the analysis and figures <a
href="01_GTEx-analysis.html">here</a>.</p>
</div>
<div id="simulation-examples" class="section level2">
<h2>3. Simulation examples</h2>
<p>See the simulations, analysis, and figures <a
href="02_simulation-analysis.html">here</a>.</p>
<hr />
<p><a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png" /></a><br />This
work is licensed under a
<a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/">Creative
Commons Attribution-ShareAlike 4.0 International License</a>.</p>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
