<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="C.B. Del Azodi" />

<meta name="date" content="2025-01-24" />

<title>QTLExperiment and multistateQTL: GTEx analysis</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">WEEO_2021_multi-QTL-benchmark</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="01_GTEx-analysis.html">GTEx Analysis</a>
</li>
<li>
  <a href="02_simulation-analysis.html">Simulation Analysis</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2021_meta-eQTL">
    <span class="fa fa-gitlab"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">QTLExperiment and multistateQTL: GTEx
analysis</h1>
<h4 class="author">C.B. Del Azodi</h4>
<h4 class="date">2025-01-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>QTLE_Manuscript_results/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190102code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20190102)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190102code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190102)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomdunstoneaQTLEManuscriptresultstreeb8a76e33266f9a634fb449c549777c0c5901f3d0targetblankb8a76e3a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/dunstone-a/QTLE_Manuscript_results/tree/b8a76e33266f9a634fb449c549777c0c5901f3d0" target="_blank">b8a76e3</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomdunstoneaQTLEManuscriptresultstreeb8a76e33266f9a634fb449c549777c0c5901f3d0targetblankb8a76e3a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/dunstone-a/QTLE_Manuscript_results/tree/b8a76e33266f9a634fb449c549777c0c5901f3d0" target="_blank">b8a76e3</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/Adipose_Subcutaneous.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Artery_Tibial.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Esophagus_Mucosa.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Lung.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Muscle_Skeletal.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Nerve_Tibial.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Skin_Not_Sun_Exposed_Suprapubic.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Skin_Sun_Exposed_Lower_leg.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Thyroid.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Whole_Blood.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    output/00_Figures/
    Ignored:    output/gtex10_chr1_QTLE.rds
    Ignored:    output/gtex10_chr1_QTLE_filt.rds
    Ignored:    output/gtex_estimated_params.rds

Untracked files:
    Untracked:  QTLE_Manuscript_results.Rproj
    Untracked:  analysis/01_GTEx-analysis_old.Rmd
    Untracked:  analysis/about_old.Rmd

Unstaged changes:
    Modified:   analysis/01_GTEx-analysis.Rmd
    Modified:   analysis/_site.yml
    Modified:   code/preprocess_gtex_alltests.sh
    Modified:   data/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="set-up-environment" class="section level2">
<h2>Set up environment</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>  <span class="fu">library</span>(ggpubr)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>  <span class="fu">library</span>(dplyr)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>  <span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>  <span class="fu">library</span>(cowplot)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>  <span class="fu">library</span>(QTLExperiment)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>  devtools<span class="sc">::</span><span class="fu">load_all</span>(<span class="st">&quot;/mnt/beegfs/mccarthy/scratch/general/adunstone/CBD_2021_qtlexperiment&quot;</span>)</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>  devtools<span class="sc">::</span><span class="fu">load_all</span>(<span class="st">&quot;/mnt/beegfs/mccarthy/scratch/general/adunstone/CBD_2021_multistateQTL&quot;</span>)</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>})</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>rerun <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a>plot_theme <span class="ot">&lt;-</span> <span class="fu">theme_cowplot</span>(<span class="dv">8</span>)</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>save <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a>save.loc <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;output/00_Figures/&quot;</span>, <span class="fu">Sys.Date</span>(), <span class="st">&quot;_&quot;</span>)</span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;output/00_Figures&quot;</span>)</span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a><span class="co"># Simulation parameters</span></span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a>n_tests <span class="ot">&lt;-</span> <span class="dv">500</span></span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a>n_states <span class="ot">&lt;-</span> <span class="dv">6</span></span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a>thresh <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a>second_thresh <span class="ot">&lt;-</span> <span class="fl">0.2</span></span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a>factor <span class="ot">&lt;-</span> <span class="fl">0.75</span></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a>sharing <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="gtex-analysis" class="section level1">
<h1>GTEx analysis</h1>
<div id="load-empirical-data" class="section level2">
<h2>Load empirical data</h2>
<p>For this study and to set default simulation parameters for
<code>qtleParams()</code>, GTEx v8 eQTL summary statistics from 10
tissues with the largest sample size for eQTL mapping:</p>
<ol style="list-style-type: decimal">
<li>Muscle_Skeletal : 706</li>
<li>Whole_Blood : 670</li>
<li>Skin_Sun_Exposed_Lower_leg : 605</li>
<li>Artery_Tibial : 584</li>
<li>Adipose_Subcutaneous : 581</li>
<li>Thyroid : 574</li>
<li>Nerve_Tibial : 532</li>
<li>Skin_Not_Sun_Exposed_Suprapubic : 517</li>
<li>Lung : 515</li>
<li>Esophagus_Mucosa : 497</li>
</ol>
<p>Data were downloaded from <a
href="https://console.cloud.google.com/storage/browser/gtex-resources/GTEx_Analysis_v8_QTLs/GTEx_Analysis_v8_eQTL_all_associations?pageState=(%22StorageObjectListTable%22%3A(%22f%22%3A%22%255B%255D%22))&amp;prefix&amp;forceOnObjectsSortingFiltering=false&amp;cloudshell=true">Google
Cloud</a>). To speed up calculations, only associations on chromosome 1
were used. The threshold for tests used to estimate significant
parameters was &lt;0.01 and for null QTL parameters was &gt;0.1.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="cf">if</span>(rerun){</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>  wd <span class="ot">&lt;-</span> <span class="st">&quot;data/&quot;</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  file_id <span class="ot">&lt;-</span> <span class="st">&quot;.allpairs.txt.gz_alltests_chr1.txt&quot;</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  input_hpc <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Lung =</span> <span class="fu">paste0</span>(wd, <span class="st">&quot;Lung&quot;</span>, file_id),</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>                    <span class="at">Whole_Blood =</span> <span class="fu">paste0</span>(wd, <span class="st">&quot;Whole_Blood&quot;</span>, file_id),</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>                    <span class="at">Thyroid =</span> <span class="fu">paste0</span>(wd, <span class="st">&quot;Thyroid&quot;</span>, file_id),</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>                    <span class="at">Skin_nosun =</span> <span class="fu">paste0</span>(wd, <span class="st">&quot;Skin_Not_Sun_Exposed_Suprapubic&quot;</span>, file_id),</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>                    <span class="at">Skin_sun =</span> <span class="fu">paste0</span>(wd, <span class="st">&quot;Skin_Sun_Exposed_Lower_leg&quot;</span>, file_id),</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>                    <span class="at">Nerve_Tibial =</span> <span class="fu">paste0</span>(wd, <span class="st">&quot;Nerve_Tibial&quot;</span>, file_id),</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>                    <span class="at">Muscle_Skeletal =</span> <span class="fu">paste0</span>(wd, <span class="st">&quot;Muscle_Skeletal&quot;</span>, file_id),</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>                    <span class="at">Adipose_Subcutaneous =</span> <span class="fu">paste0</span>(wd, <span class="st">&quot;Adipose_Subcutaneous&quot;</span>, file_id),</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>                    <span class="at">Artery_Tibial =</span> <span class="fu">paste0</span>(wd, <span class="st">&quot;Artery_Tibial&quot;</span>, file_id),</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>                    <span class="at">Esophagus_Mucosa =</span> <span class="fu">paste0</span>(wd, <span class="st">&quot;Esophagus_Mucosa&quot;</span>, file_id))</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>  </span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>  <span class="fu">system.time</span>(gtex <span class="ot">&lt;-</span> <span class="fu">sumstats2qtle</span>(input_hpc, </span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>                                    <span class="at">feature_id=</span><span class="st">&quot;gene_id&quot;</span>,</span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a>                                    <span class="at">variant_id=</span><span class="st">&quot;variant_id&quot;</span>, </span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a>                                    <span class="at">betas =</span> <span class="st">&quot;slope&quot;</span>, </span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a>                                    <span class="at">errors =</span> <span class="st">&quot;slope_se&quot;</span>,</span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a>                                    <span class="at">pvalues =</span> <span class="st">&quot;pval_nominal&quot;</span>))</span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a>  <span class="fu">saveRDS</span>(gtex, <span class="st">&quot;output/gtex10_chr1_QTLE.rds&quot;</span>)</span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a>  </span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Filtering to remove tests with NAs in any tissue:&quot;</span>)</span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a>  <span class="fu">system.time</span>(gtex <span class="ot">&lt;-</span> <span class="fu">getComplete</span>(gtex, <span class="at">verbose=</span><span class="cn">TRUE</span>))</span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a>  <span class="fu">saveRDS</span>(gtex, <span class="st">&quot;output/gtex10_chr1_QTLE_filt.rds&quot;</span>)</span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a>  </span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a>} <span class="cf">else</span>{</span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a>  gtex <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;output/gtex10_chr1_QTLE_filt.rds&quot;</span>)</span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a>}</span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a></span>
<span id="cb2-31"><a href="#cb2-31" tabindex="-1"></a>gtex</span></code></pre></div>
<pre><code>class: QTLExperiment 
dim: 11680281 10 
metadata(0):
assays(3): betas errors pvalues
rownames(11680281): ENSG00000227232.5|chr1_13550_G_A_b38
  ENSG00000227232.5|chr1_14671_G_C_b38 ...
  ENSG00000185220.11|chr1_248944397_A_C_b38
  ENSG00000185220.11|chr1_248945104_C_T_b38
rowData names(2): variant_id feature_id
colnames(10): Lung Whole_Blood ... Artery_Tibial Esophagus_Mucosa
colData names(1): state_id</code></pre>
<p>Note it took 637.990 seconds (10 min 38 sec) to load 20,177,150 QTL
for 10 tissues. Note that 11,680,281 QTL remain after removing those
with NAs in any of the 10 tissues.</p>
</div>
<div id="call-significant-and-top-qtl" class="section level2">
<h2>Call significant and top QTL</h2>
<p>Next we call QTL as significant if it has a p.value &lt; 0.01 in at
least one tissue and &lt; 0.05 in any other tissues using
<code>callSignificance()</code>. We then filter to keep only the most
significant QTL for each feature (i.e. gene) for each tissue using
<code>getTopHits()</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">system.time</span>(gtex <span class="ot">&lt;-</span> <span class="fu">callSignificance</span>(gtex, </span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>                                     <span class="at">assay=</span><span class="st">&quot;pvalues&quot;</span>,</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>                                     <span class="at">thresh=</span>thresh, </span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>                                     <span class="at">second.thresh=</span>second_thresh))</span></code></pre></div>
<pre><code>   user  system elapsed 
  7.690   0.560   8.289 </code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">system.time</span>(gtex_top <span class="ot">&lt;-</span> <span class="fu">getTopHits</span>(gtex, <span class="at">mode=</span><span class="st">&quot;state&quot;</span>))</span></code></pre></div>
<pre><code>   user  system elapsed 
 22.346   1.911  24.351 </code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">table</span>(<span class="fu">feature_id</span>(gtex_top)), <span class="at">xlab =</span> <span class="st">&quot;# top QTL per feature&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figure/01_GTEx-analysis_old.Rmd/get-top-hits-stats-1.png" alt="The distribution of the number of top-QTL hits for each feature, where 10 means a different QTL was the top hit in each tissue." width="768" />
<p class="caption">
The distribution of the number of top-QTL hits for each feature, where
10 means a different QTL was the top hit in each tissue.
</p>
</div>
</div>
<div id="global-patterns-of-multi-tissue-qtl" class="section level2">
<h2>Global patterns of multi-tissue QTL</h2>
<p>With significant QTL called, we can next calculate the degree of
pairwise sharing of eQTL across tissues using the
<code>runPairwiseSharing()</code> function and visualize the pattern of
sharing with <code>plotPairwiseSharing()</code></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">system.time</span>(gtex_top <span class="ot">&lt;-</span> <span class="fu">runPairwiseSharing</span>(gtex_top, </span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>                                           <span class="at">assay=</span><span class="st">&quot;significant&quot;</span>, </span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>                                           <span class="at">factor=</span>factor))</span></code></pre></div>
<pre><code>   user  system elapsed 
  0.096   0.000   0.096 </code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">system.time</span>(p_gtex <span class="ot">&lt;-</span> <span class="fu">plotPairwiseSharing</span>(gtex_top, </span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>                                          <span class="at">cell_annotate =</span> <span class="cn">TRUE</span>, </span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>                                          <span class="at">col_range =</span> sharing))</span></code></pre></div>
<pre><code>   user  system elapsed 
  0.079   0.002   0.082 </code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">draw</span>(p_gtex)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figure/01_GTEx-analysis_old.Rmd/plot-gtex-pairwise-sharing-1.png" alt="Pairwise sharing of GTEx top-eQTL." width="576" />
<p class="caption">
Pairwise sharing of GTEx top-eQTL.
</p>
</div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="cf">if</span>(save){</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">paste0</span>(save.loc, <span class="st">&quot;gtex_pairwise.pdf&quot;</span>)</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Saving to: &quot;</span>, tmp)</span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>  <span class="fu">pdf</span>(tmp, <span class="at">height=</span><span class="dv">6</span>, <span class="at">width=</span><span class="fl">6.5</span>)</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>  <span class="fu">draw</span>(p_gtex)</span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>  <span class="cf">while</span> (<span class="sc">!</span><span class="fu">is.null</span>(<span class="fu">dev.list</span>()))  <span class="fu">dev.off</span>()</span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Comparing skin sun vs skin no-sun:&quot;</span>)</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">plotCompareStates</span>(gtex_top, <span class="at">x=</span><span class="st">&quot;Skin_sun&quot;</span>, <span class="at">y=</span><span class="st">&quot;Skin_nosun&quot;</span>)</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>p1<span class="sc">$</span>counts</span></code></pre></div>
<pre><code>
both_diverging    both_shared        not_sig     Skin_nosun       Skin_sun 
           152           6393           4088           1905           2223 </code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> p1<span class="sc">$</span>plot <span class="sc">+</span> <span class="fu">theme_cowplot</span>()</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Comparing skin sun vs whole blood:&quot;</span>)</span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">plotCompareStates</span>(gtex_top, <span class="at">x=</span><span class="st">&quot;Skin_sun&quot;</span>, <span class="at">y=</span><span class="st">&quot;Whole_Blood&quot;</span>)</span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a>p2<span class="sc">$</span>counts</span></code></pre></div>
<pre><code>
both_diverging    both_shared        not_sig       Skin_sun    Whole_Blood 
           921           4610           3665           3237           2328 </code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> p2<span class="sc">$</span>plot <span class="sc">+</span> <span class="fu">theme_cowplot</span>()</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a><span class="fu">plot_grid</span>(p1 <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>), p2, </span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a>          <span class="at">rel_widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.6</span>), <span class="at">labels =</span> <span class="st">&quot;auto&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figure/01_GTEx-analysis_old.Rmd/plot-gtex-comparisons-1.png" alt="Comparison of estimated effect sizes of top-eQTL in (a) skin sun vs skin no-sun and (b) skin sun vs whole blood." width="672" />
<p class="caption">
Comparison of estimated effect sizes of top-eQTL in (a) skin sun vs skin
no-sun and (b) skin sun vs whole blood.
</p>
</div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="cf">if</span>(save){</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="fu">paste0</span>(save.loc, <span class="st">&quot;GTEX_compare-states.pdf&quot;</span>), <span class="fu">last_plot</span>(), <span class="at">height =</span> <span class="dv">4</span>, <span class="at">width =</span> <span class="dv">7</span>)</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="inspecting-multi-tissue-qtl" class="section level2">
<h2>Inspecting multi-tissue QTL</h2>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>gtex_top <span class="ot">&lt;-</span> <span class="fu">runTestMetrics</span>(gtex_top, <span class="at">global_buffer =</span> <span class="dv">2</span>)</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">rowData</span>(gtex_top)<span class="sc">$</span>qtl_type_simple)</span></code></pre></div>
<pre><code>
    global multistate     unique 
      5169       8880        712 </code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>gtex_top_multistate <span class="ot">&lt;-</span> <span class="fu">subset</span>(gtex_top, nSignificant <span class="sc">&gt;</span> <span class="dv">4</span> <span class="sc">&amp;</span> nSignificant <span class="sc">&lt;=</span> <span class="dv">8</span>)</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>p_qtl <span class="ot">&lt;-</span> <span class="fu">plotQTLClusters</span>(gtex_top_multistate, <span class="at">FUN=</span>abs, <span class="at">row_km =</span> <span class="dv">5</span>, </span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>                         <span class="at">annotate_tests =</span> <span class="fu">c</span>(<span class="st">&quot;effect_sd&quot;</span>, <span class="st">&quot;nSignificant&quot;</span>))</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a><span class="fu">draw</span>(p_qtl)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figure/01_GTEx-analysis_old.Rmd/plot-GTEx-tissue-specific-1.png" alt="The absolute values of estimated effect sizes for a subset of multi-state QTL." width="384" />
<p class="caption">
The absolute values of estimated effect sizes for a subset of
multi-state QTL.
</p>
</div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="cf">if</span>(save){</span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">paste0</span>(save.loc, <span class="st">&quot;sim_simple_pairwise.pdf&quot;</span>)</span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Saving to: &quot;</span>, tmp)</span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>  <span class="fu">pdf</span>(tmp, <span class="at">height=</span><span class="dv">5</span>, <span class="at">width=</span><span class="dv">4</span>)</span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a>  <span class="fu">draw</span>(p_qtl)</span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a>  <span class="cf">while</span> (<span class="sc">!</span><span class="fu">is.null</span>(<span class="fu">dev.list</span>()))  <span class="fu">dev.off</span>()</span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="estimating-simulation-parameters" class="section level2">
<h2>Estimating simulation parameters</h2>
<p>Using <code>qtleEstimate()</code>, simulation parameters are
estimated from the provided GTEx data. The GTEx summary statistics with
p values in the lowest quantile are used to estimate parameters to
significant tests and those in the largest quantile are used to estimate
parameters for null tests.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>params <span class="ot">&lt;-</span> <span class="fu">qtleEstimate</span>(gtex, <span class="at">assay =</span> <span class="st">&quot;pvalues&quot;</span>, <span class="at">verbose=</span><span class="cn">TRUE</span>,</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>                       <span class="at">thresh_sig=</span><span class="fl">0.001</span>, </span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>                       <span class="at">thresh_null=</span><span class="fl">0.1</span>)</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a><span class="cf">if</span>(save){</span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a>  <span class="fu">saveRDS</span>(params, <span class="st">&quot;output/gtex_estimated_params.rds&quot;</span>)</span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a>}</span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a>params</span></code></pre></div>
<pre><code>$cv.sig.shape
[1] 13.59969

$cv.sig.rate
[1] 0.03282103

$cv.null.shape
[1] 3.995155e-05

$cv.null.rate
[1] 476510.7

$betas.sig.shape
[1] 2.160491

$betas.sig.rate
[1] 14.19135

$betas.null.shape
[1] 1.053145

$betas.null.rate
[1] 22.55344</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="fu">plotSimulationParams</span>(<span class="at">params=</span>params) <span class="sc">+</span></span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">&quot;Dark2&quot;</span>) <span class="sc">+</span> </span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">&quot;Dark2&quot;</span>) <span class="sc">+</span> </span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a>  plot_theme</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figure/01_GTEx-analysis_old.Rmd/plot-estimated%20params-1.png" alt="Distribution of sampled betas and coefficient of variation values for significant and null tests using the parameters estimated from GTEx." width="576" />
<p class="caption">
Distribution of sampled betas and coefficient of variation values for
significant and null tests using the parameters estimated from GTEx.
</p>
</div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="cf">if</span>(save){</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="fu">paste0</span>(save.loc, <span class="st">&quot;GTEX_estParam.pdf&quot;</span>), <span class="fu">last_plot</span>(), <span class="at">height =</span> <span class="dv">4</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a>}</span></code></pre></div>
<p>These parameters are used as the default parameters with
<code>qtleParams()</code>.</p>
</div>
</div>
<div id="session-info" class="section level1">
<h1>Session Info</h1>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Rocky Linux 8.10 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.15.so;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Melbourne
tzcode source: system (glibc)

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] multiStateQTL_0.1.0         collapse_2.0.16            
 [3] data.table_1.16.0           QTLExperiment_0.1.0        
 [5] testthat_3.2.1.1            SummarizedExperiment_1.35.2
 [7] Biobase_2.65.1              GenomicRanges_1.57.1       
 [9] GenomeInfoDb_1.41.1         IRanges_2.39.2             
[11] S4Vectors_0.43.2            BiocGenerics_0.51.2        
[13] MatrixGenerics_1.17.0       matrixStats_1.4.1          
[15] cowplot_1.1.3               ComplexHeatmap_2.21.1      
[17] dplyr_1.1.4                 ggpubr_0.6.0               
[19] ggplot2_3.5.1              

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      rstudioapi_0.16.0       jsonlite_1.8.9         
  [4] shape_1.4.6.1           magrittr_2.0.3          farver_2.1.2           
  [7] rmarkdown_2.28          ragg_1.3.3              GlobalOptions_0.1.2    
 [10] fs_1.6.4                zlibbioc_1.51.1         vctrs_0.6.5            
 [13] memoise_2.0.1           SQUAREM_2021.1          rstatix_0.7.2          
 [16] mixsqp_0.3-54           htmltools_0.5.8.1       S4Arrays_1.5.10        
 [19] usethis_3.0.0           truncnorm_1.0-9         broom_1.0.7            
 [22] SparseArray_1.5.41      Formula_1.2-5           mashr_0.2.79           
 [25] sass_0.4.9              bslib_0.8.0             htmlwidgets_1.6.4      
 [28] desc_1.4.3              plyr_1.8.9              cachem_1.1.0           
 [31] mime_0.12               lifecycle_1.0.4         iterators_1.0.14       
 [34] pkgconfig_2.0.3         Matrix_1.7-0            R6_2.5.1               
 [37] fastmap_1.2.0           GenomeInfoDbData_1.2.13 fitdistrplus_1.2-1     
 [40] shiny_1.9.1             clue_0.3-65             digest_0.6.37          
 [43] colorspace_2.1-1        rprojroot_2.0.4         irlba_2.3.5.1          
 [46] pkgload_1.4.0           textshaping_0.4.0       labeling_0.4.3         
 [49] invgamma_1.1            fansi_1.0.6             httr_1.4.7             
 [52] abind_1.4-8             compiler_4.4.1          remotes_2.5.0          
 [55] bit64_4.5.2             withr_3.0.1             doParallel_1.0.17      
 [58] backports_1.5.0         viridis_0.6.5           carData_3.0-5          
 [61] highr_0.11              pkgbuild_1.4.4          ggsignif_0.6.4         
 [64] MASS_7.3-61             DelayedArray_0.31.12    sessioninfo_1.2.2      
 [67] rjson_0.2.23            tools_4.4.1             httpuv_1.6.15          
 [70] glue_1.8.0              promises_1.3.0          checkmate_2.3.2        
 [73] cluster_2.1.6           generics_0.1.3          gtable_0.3.5           
 [76] tzdb_0.4.0              tidyr_1.3.1             car_3.1-3              
 [79] utf8_1.2.4              XVector_0.45.0          foreach_1.5.2          
 [82] pillar_1.9.0            stringr_1.5.1           vroom_1.6.5            
 [85] later_1.3.2             splines_4.4.1           circlize_0.4.16        
 [88] lattice_0.22-6          survival_3.7-0          bit_4.5.0              
 [91] tidyselect_1.2.1        miniUI_0.1.1.1          knitr_1.48             
 [94] git2r_0.33.0            gridExtra_2.3           xfun_0.47              
 [97] devtools_2.4.5          brio_1.1.5              stringi_1.8.4          
[100] UCSC.utils_1.1.0        workflowr_1.7.1         yaml_2.3.10            
[103] evaluate_1.0.0          codetools_0.2-20        tibble_3.2.1           
[106] cli_3.6.3               systemfonts_1.1.0       xtable_1.8-4           
[109] munsell_0.5.1           jquerylib_0.1.4         Rcpp_1.0.13            
[112] rmeta_3.0               png_0.1-8               parallel_4.4.1         
[115] ellipsis_0.3.2          assertthat_0.2.1        profvis_0.4.0          
[118] urlchecker_1.0.1        mvtnorm_1.3-1           viridisLite_0.4.2      
[121] scales_1.3.0            purrr_1.0.2             crayon_1.5.3           
[124] GetoptLong_1.0.5        rlang_1.1.4             ashr_2.2-63            </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.4.1 (2024-06-14)
 os       Rocky Linux 8.10 (Green Obsidian)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_AU.UTF-8
 ctype    en_AU.UTF-8
 tz       Australia/Melbourne
 date     2025-01-24
 pandoc   3.1.1 @ /usr/lib/rstudio-server/bin/quarto/bin/tools/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 !  package              * version date (UTC) lib source
    abind                  1.4-8   2024-09-12 [1] CRAN (R 4.4.1)
    ashr                   2.2-63  2023-08-21 [1] CRAN (R 4.4.1)
    assertthat             0.2.1   2019-03-21 [1] CRAN (R 4.4.1)
    backports              1.5.0   2024-05-23 [1] CRAN (R 4.4.1)
    Biobase              * 2.65.1  2024-08-28 [1] Bioconductor 3.20 (R 4.4.1)
    BiocGenerics         * 0.51.2  2024-09-27 [1] Bioconductor 3.20 (R 4.4.1)
    bit                    4.5.0   2024-09-20 [1] CRAN (R 4.4.1)
    bit64                  4.5.2   2024-09-22 [1] CRAN (R 4.4.1)
    brio                   1.1.5   2024-04-24 [1] CRAN (R 4.4.1)
    broom                  1.0.7   2024-09-26 [1] CRAN (R 4.4.1)
    bslib                  0.8.0   2024-07-29 [1] CRAN (R 4.4.1)
    cachem                 1.1.0   2024-05-16 [1] CRAN (R 4.4.1)
    car                    3.1-3   2024-09-27 [1] CRAN (R 4.4.1)
    carData                3.0-5   2022-01-06 [1] CRAN (R 4.4.1)
    checkmate              2.3.2   2024-07-29 [1] CRAN (R 4.4.1)
    circlize               0.4.16  2024-02-20 [1] CRAN (R 4.4.1)
    cli                    3.6.3   2024-06-21 [1] CRAN (R 4.4.1)
    clue                   0.3-65  2023-09-23 [1] CRAN (R 4.4.1)
    cluster                2.1.6   2023-12-01 [2] CRAN (R 4.4.1)
    codetools              0.2-20  2024-03-31 [2] CRAN (R 4.4.1)
    collapse             * 2.0.16  2024-08-21 [1] CRAN (R 4.4.1)
    colorspace             2.1-1   2024-07-26 [1] CRAN (R 4.4.1)
    ComplexHeatmap       * 2.21.1  2024-09-24 [1] Bioconductor 3.20 (R 4.4.1)
    cowplot              * 1.1.3   2024-01-22 [1] CRAN (R 4.4.1)
    crayon                 1.5.3   2024-06-20 [1] CRAN (R 4.4.1)
    data.table           * 1.16.0  2024-08-27 [1] CRAN (R 4.4.1)
    DelayedArray           0.31.12 2024-09-27 [1] Bioconductor 3.20 (R 4.4.1)
    desc                   1.4.3   2023-12-10 [1] CRAN (R 4.4.1)
    devtools               2.4.5   2022-10-11 [1] CRAN (R 4.4.1)
    digest                 0.6.37  2024-08-19 [1] CRAN (R 4.4.1)
    doParallel             1.0.17  2022-02-07 [1] CRAN (R 4.4.1)
    dplyr                * 1.1.4   2023-11-17 [1] CRAN (R 4.4.1)
    ellipsis               0.3.2   2021-04-29 [1] CRAN (R 4.4.1)
    evaluate               1.0.0   2024-09-17 [1] CRAN (R 4.4.1)
    fansi                  1.0.6   2023-12-08 [1] CRAN (R 4.4.1)
    farver                 2.1.2   2024-05-13 [1] CRAN (R 4.4.1)
    fastmap                1.2.0   2024-05-15 [1] CRAN (R 4.4.1)
    fitdistrplus           1.2-1   2024-07-12 [1] CRAN (R 4.4.1)
    foreach                1.5.2   2022-02-02 [1] CRAN (R 4.4.1)
    Formula                1.2-5   2023-02-24 [1] CRAN (R 4.4.1)
    fs                     1.6.4   2024-04-25 [1] CRAN (R 4.4.1)
    generics               0.1.3   2022-07-05 [1] CRAN (R 4.4.1)
    GenomeInfoDb         * 1.41.1  2024-05-24 [1] Bioconductor 3.20 (R 4.4.1)
    GenomeInfoDbData       1.2.13  2024-10-02 [1] Bioconductor
    GenomicRanges        * 1.57.1  2024-06-12 [1] Bioconductor 3.20 (R 4.4.1)
    GetoptLong             1.0.5   2020-12-15 [1] CRAN (R 4.4.1)
    ggplot2              * 3.5.1   2024-04-23 [1] CRAN (R 4.4.1)
    ggpubr               * 0.6.0   2023-02-10 [1] CRAN (R 4.4.1)
    ggsignif               0.6.4   2022-10-13 [1] CRAN (R 4.4.1)
    git2r                  0.33.0  2023-11-26 [1] CRAN (R 4.4.1)
    GlobalOptions          0.1.2   2020-06-10 [1] CRAN (R 4.4.1)
    glue                   1.8.0   2024-09-30 [1] CRAN (R 4.4.1)
    gridExtra              2.3     2017-09-09 [1] CRAN (R 4.4.1)
    gtable                 0.3.5   2024-04-22 [1] CRAN (R 4.4.1)
    highr                  0.11    2024-05-26 [1] CRAN (R 4.4.1)
    htmltools              0.5.8.1 2024-04-04 [1] CRAN (R 4.4.1)
    htmlwidgets            1.6.4   2023-12-06 [1] CRAN (R 4.4.1)
    httpuv                 1.6.15  2024-03-26 [1] CRAN (R 4.4.1)
    httr                   1.4.7   2023-08-15 [1] CRAN (R 4.4.1)
    invgamma               1.1     2017-05-07 [1] CRAN (R 4.4.1)
    IRanges              * 2.39.2  2024-07-17 [1] Bioconductor 3.20 (R 4.4.1)
    irlba                  2.3.5.1 2022-10-03 [1] CRAN (R 4.4.1)
    iterators              1.0.14  2022-02-05 [1] CRAN (R 4.4.1)
    jquerylib              0.1.4   2021-04-26 [1] CRAN (R 4.4.1)
    jsonlite               1.8.9   2024-09-20 [1] CRAN (R 4.4.1)
    knitr                  1.48    2024-07-07 [1] CRAN (R 4.4.1)
    labeling               0.4.3   2023-08-29 [1] CRAN (R 4.4.1)
    later                  1.3.2   2023-12-06 [1] CRAN (R 4.4.1)
    lattice                0.22-6  2024-03-20 [2] CRAN (R 4.4.1)
    lifecycle              1.0.4   2023-11-07 [1] CRAN (R 4.4.1)
    magrittr               2.0.3   2022-03-30 [1] CRAN (R 4.4.1)
    mashr                  0.2.79  2023-10-18 [1] CRAN (R 4.4.1)
    MASS                   7.3-61  2024-06-13 [1] CRAN (R 4.4.1)
    Matrix                 1.7-0   2024-04-26 [2] CRAN (R 4.4.1)
    MatrixGenerics       * 1.17.0  2024-05-01 [1] Bioconductor 3.20 (R 4.4.1)
    matrixStats          * 1.4.1   2024-09-08 [1] CRAN (R 4.4.1)
    memoise                2.0.1   2021-11-26 [1] CRAN (R 4.4.1)
    mime                   0.12    2021-09-28 [1] CRAN (R 4.4.1)
    miniUI                 0.1.1.1 2018-05-18 [1] CRAN (R 4.4.1)
    mixsqp                 0.3-54  2023-12-20 [1] CRAN (R 4.4.1)
 R  multiStateQTL        * 0.1.0   &lt;NA&gt;       [?] &lt;NA&gt;
    munsell                0.5.1   2024-04-01 [1] CRAN (R 4.4.1)
    mvtnorm                1.3-1   2024-09-03 [1] CRAN (R 4.4.1)
    pillar                 1.9.0   2023-03-22 [1] CRAN (R 4.4.1)
    pkgbuild               1.4.4   2024-03-17 [1] CRAN (R 4.4.1)
    pkgconfig              2.0.3   2019-09-22 [1] CRAN (R 4.4.1)
    pkgload                1.4.0   2024-06-28 [1] CRAN (R 4.4.1)
    plyr                   1.8.9   2023-10-02 [1] CRAN (R 4.4.1)
    png                    0.1-8   2022-11-29 [1] CRAN (R 4.4.1)
    profvis                0.4.0   2024-09-20 [1] CRAN (R 4.4.1)
    promises               1.3.0   2024-04-05 [1] CRAN (R 4.4.1)
    purrr                  1.0.2   2023-08-10 [1] CRAN (R 4.4.1)
 VP QTLExperiment        * 0.1.0   2024-10-05 [?] git2r (https://github.com/dunstone-a/QTLExperiment.git@4c1b6f8) (on disk 1.3.0)
    R6                     2.5.1   2021-08-19 [1] CRAN (R 4.4.1)
    ragg                   1.3.3   2024-09-11 [1] CRAN (R 4.4.1)
    RColorBrewer           1.1-3   2022-04-03 [1] CRAN (R 4.4.1)
    Rcpp                   1.0.13  2024-07-17 [1] CRAN (R 4.4.1)
    remotes                2.5.0   2024-03-17 [1] CRAN (R 4.4.1)
    rjson                  0.2.23  2024-09-16 [1] CRAN (R 4.4.1)
    rlang                  1.1.4   2024-06-04 [1] CRAN (R 4.4.1)
    rmarkdown              2.28    2024-08-17 [1] CRAN (R 4.4.1)
    rmeta                  3.0     2018-03-20 [1] CRAN (R 4.4.1)
    rprojroot              2.0.4   2023-11-05 [1] CRAN (R 4.4.1)
    rstatix                0.7.2   2023-02-01 [1] CRAN (R 4.4.1)
    rstudioapi             0.16.0  2024-03-24 [1] CRAN (R 4.4.1)
    S4Arrays               1.5.10  2024-09-29 [1] Bioconductor 3.20 (R 4.4.1)
    S4Vectors            * 0.43.2  2024-07-17 [1] Bioconductor 3.20 (R 4.4.1)
    sass                   0.4.9   2024-03-15 [1] CRAN (R 4.4.1)
    scales                 1.3.0   2023-11-28 [1] CRAN (R 4.4.1)
    sessioninfo            1.2.2   2021-12-06 [1] CRAN (R 4.4.1)
    shape                  1.4.6.1 2024-02-23 [1] CRAN (R 4.4.1)
    shiny                  1.9.1   2024-08-01 [1] CRAN (R 4.4.1)
    SparseArray            1.5.41  2024-09-27 [1] Bioconductor 3.20 (R 4.4.1)
    SQUAREM                2021.1  2021-01-13 [1] CRAN (R 4.4.1)
    stringi                1.8.4   2024-05-06 [1] CRAN (R 4.4.1)
    stringr                1.5.1   2023-11-14 [1] CRAN (R 4.4.1)
    SummarizedExperiment * 1.35.2  2024-09-27 [1] Bioconductor 3.20 (R 4.4.1)
    survival               3.7-0   2024-06-05 [1] CRAN (R 4.4.1)
    systemfonts            1.1.0   2024-05-15 [1] CRAN (R 4.4.1)
    testthat             * 3.2.1.1 2024-04-14 [1] CRAN (R 4.4.1)
    textshaping            0.4.0   2024-05-24 [1] CRAN (R 4.4.1)
    tibble                 3.2.1   2023-03-20 [1] CRAN (R 4.4.1)
    tidyr                  1.3.1   2024-01-24 [1] CRAN (R 4.4.1)
    tidyselect             1.2.1   2024-03-11 [1] CRAN (R 4.4.1)
    truncnorm              1.0-9   2023-03-20 [1] CRAN (R 4.4.1)
    tzdb                   0.4.0   2023-05-12 [1] CRAN (R 4.4.1)
    UCSC.utils             1.1.0   2024-05-01 [1] Bioconductor 3.20 (R 4.4.1)
    urlchecker             1.0.1   2021-11-30 [1] CRAN (R 4.4.1)
    usethis                3.0.0   2024-07-29 [1] CRAN (R 4.4.1)
    utf8                   1.2.4   2023-10-22 [1] CRAN (R 4.4.1)
    vctrs                  0.6.5   2023-12-01 [1] CRAN (R 4.4.1)
    viridis                0.6.5   2024-01-29 [1] CRAN (R 4.4.1)
    viridisLite            0.4.2   2023-05-02 [1] CRAN (R 4.4.1)
    vroom                  1.6.5   2023-12-05 [1] CRAN (R 4.4.1)
    withr                  3.0.1   2024-07-31 [1] CRAN (R 4.4.1)
    workflowr              1.7.1   2023-08-23 [1] CRAN (R 4.4.1)
    xfun                   0.47    2024-08-17 [1] CRAN (R 4.4.1)
    xtable                 1.8-4   2019-04-21 [1] CRAN (R 4.4.1)
    XVector                0.45.0  2024-05-01 [1] Bioconductor 3.20 (R 4.4.1)
    yaml                   2.3.10  2024-07-26 [1] CRAN (R 4.4.1)
    zlibbioc               1.51.1  2024-06-05 [1] Bioconductor 3.20 (R 4.4.1)

 [1] /mnt/beegfs/users/amelia.dunstone/R/x86_64-pc-linux-gnu-library/4.4
 [2] /opt/R/4.4.1/lib/R/library

 V ── Loaded and on-disk version mismatch.
 P ── Loaded and on-disk path mismatch.
 R ── Package was removed from disk.

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
