<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="C.B. Del Azodi" />

<meta name="date" content="2022-11-10" />

<title>QTLExperiment and multistateQTL: Simulation examples</title>

<script src="site_libs/header-attrs-2.15/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">WEEO_2021_multi-QTL-benchmark</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2021_meta-eQTL">
    <span class="fa fa-gitlab"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">QTLExperiment and multistateQTL: Simulation
examples</h1>
<h4 class="author">C.B. Del Azodi</h4>
<h4 class="date">2022-11-10</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>WEEO_2021_meta-eQTL/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190102code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20190102)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190102code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190102)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgitlabsvieduaubiocellgenprivateWEEO2022multistateQTLtree83405215772a39a777bd477308feb9ba4b0b9f14targetblank8340521a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/tree/83405215772a39a777bd477308feb9ba4b0b9f14" target="_blank">8340521</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgitlabsvieduaubiocellgenprivateWEEO2022multistateQTLtree83405215772a39a777bd477308feb9ba4b0b9f14targetblank8340521a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2022_multistateQTL/tree/83405215772a39a777bd477308feb9ba4b0b9f14" target="_blank">8340521</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    Manuscript-Figures_files/
    Ignored:    data/Adipose_Subcutaneous.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Adipose_Subcutaneous.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Adipose_Visceral_Omentum.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Adrenal_Gland.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Artery_Aorta.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Artery_Coronary.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Artery_Tibial.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Artery_Tibial.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Amygdala.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Anterior_cingulate_cortex_BA24.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Caudate_basal_ganglia.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Cerebellar_Hemisphere.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Cerebellum.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Cortex.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Frontal_Cortex_BA9.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Hippocampus.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Hypothalamus.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Nucleus_accumbens_basal_ganglia.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Putamen_basal_ganglia.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Spinal_cord_cervical_c-1.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Brain_Substantia_nigra.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Breast_Mammary_Tissue.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Cells_Cultured_fibroblasts.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Cells_EBV-transformed_lymphocytes.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Colon_Sigmoid.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Colon_Transverse.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Esophagus_Gastroesophageal_Junction.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Esophagus_Mucosa.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Esophagus_Mucosa.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Esophagus_Muscularis.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/GTEx_Analysis_v8_eQTL_all_associations/
    Ignored:    data/Heart_Atrial_Appendage.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Heart_Left_Ventricle.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Kidney_Cortex.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Liver.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Lung.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Lung.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Lung.v8.signif_variant_gene_pairs_chr1.txt
    Ignored:    data/Minor_Salivary_Gland.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Muscle_Skeletal.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Muscle_Skeletal.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Nerve_Tibial.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Nerve_Tibial.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Ovary.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Pancreas.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Pituitary.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Prostate.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Skin_Not_Sun_Exposed_Suprapubic.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Skin_Not_Sun_Exposed_Suprapubic.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Skin_Sun_Exposed_Lower_leg.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Skin_Sun_Exposed_Lower_leg.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Small_Intestine_Terminal_Ileum.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Spleen.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Spleen.v8.signif_variant_gene_pairs_chr1.txt
    Ignored:    data/Stomach.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Testis.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Thyroid.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Thyroid.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Thyroid.v8.signif_variant_gene_pairs_chr1.txt
    Ignored:    data/Uterus.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Vagina.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Whole_Blood.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Whole_Blood.v8.signif_variant_gene_pairs.txt.gz_chr1.txt
    Ignored:    data/Whole_Blood.v8.signif_variant_gene_pairs_chr1.txt
    Ignored:    output/00_Figures/
    Ignored:    output/01_simulations/
    Ignored:    output/gtex10_chr1_QTLE.rds
    Ignored:    output/gtex10_chr1_QTLE_filt.rds
    Ignored:    output/gtex_estimated_params.rds
    Ignored:    references/chr2.filtered.log
    Ignored:    references/chr2.filtered.vcf
    Ignored:    references/chr2.filtered2.log
    Ignored:    references/chr2.filtered2.vcf
    Ignored:    references/chr2.filtered3.log
    Ignored:    references/chr2.filtered3.vcf
    Ignored:    references/chr2.genes.gff3
    Ignored:    references/chr2.vcf.gz

Untracked files:
    Untracked:  .gitignore
    Untracked:  WEEO_2021_meta-eQTL.Rproj
    Untracked:  analysis/02_simulation-analysis.Rmd
    Untracked:  analysis/2021-08-03_pbmc8k-simulations.Rmd
    Untracked:  analysis/figure/
    Untracked:  code/preprocess_gtex.sh
    Untracked:  site_libs/
    Untracked:  workflow/

Unstaged changes:
    Deleted:    Snakefile
    Modified:   analysis/01_GTEx-analysis.Rmd
    Modified:   analysis/2021-07-30_pbmc8k-reference-processing.Rmd
    Modified:   analysis/index.Rmd

Staged changes:
    New:        analysis/01_GTEx-analysis.Rmd
    Modified:   analysis/index.Rmd
    New:        code/preprocess_gtex_alltests.sh
    Deleted:    code/sce_plots.R
    Deleted:    code/sim_celltypes.R
    Deleted:    code/sim_celltypes.Rmd
    Deleted:    envs/myenv.yaml
    Deleted:    resources/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="set-up-environment" class="section level2">
<h2>Set up environment</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggpubr)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(dplyr)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(cowplot)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(QTLExperiment)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  devtools<span class="sc">::</span><span class="fu">load_all</span>(<span class="st">&quot;/mnt/mcfiles/cazodi/Software/multiStateQTL&quot;</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>rerun <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>plot_theme <span class="ot">&lt;-</span> <span class="fu">theme_cowplot</span>(<span class="dv">8</span>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>save <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>save.loc <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;output/00_Figures/&quot;</span>, <span class="fu">Sys.Date</span>(), <span class="st">&quot;_&quot;</span>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>params <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;output/gtex_estimated_params.rds&quot;</span>)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Simulation parameters</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>n_tests <span class="ot">&lt;-</span> <span class="dv">500</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>n_states <span class="ot">&lt;-</span> <span class="dv">6</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>thresh <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>second_thresh <span class="ot">&lt;-</span> <span class="fl">0.2</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>factor <span class="ot">&lt;-</span> <span class="fl">0.75</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>sharing <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="simulation-examples" class="section level1">
<h1>Simulation examples</h1>
<div id="simple-simulation" class="section level3">
<h3>Simple simulation</h3>
<p>Using the parameters estimated from the GTEx data, here we simulate a
data set with 80% of QTL effects shared across all states and the
remaining 20% of QTL effects unique to a single state.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>sim_simple <span class="ot">&lt;-</span> <span class="fu">qtleSimulate</span>(<span class="at">params=</span>params, <span class="at">ntests=</span>n_tests, <span class="at">nstates=</span>n_states, </span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">global=</span><span class="fl">0.8</span>, <span class="at">unique=</span><span class="fl">0.2</span>) </span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>sim_simple <span class="ot">&lt;-</span> <span class="fu">callSignificance</span>(sim_simple, <span class="at">assay=</span><span class="st">&quot;lfsrs&quot;</span>, </span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">thresh=</span>thresh, <span class="at">second.thresh =</span> second_thresh)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>sim_simple <span class="ot">&lt;-</span> <span class="fu">runPairwiseSharing</span>(sim_simple, <span class="at">assay=</span><span class="st">&quot;significant&quot;</span>, <span class="at">factor=</span>factor)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>p_simple <span class="ot">&lt;-</span> <span class="fu">plotPairwiseSharing</span>(sim_simple, <span class="at">col_range=</span>sharing,</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>                                <span class="at">cell_annotate =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(p_simple)</span></code></pre></div>
<p><img src="figure/02_simulation-analysis.Rmd/global-QTL-simulation-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(save){</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">paste0</span>(save.loc, <span class="st">&quot;sim_simple_pairwise.pdf&quot;</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Saving to: &quot;</span>, tmp)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(tmp, <span class="at">height=</span><span class="dv">5</span>, <span class="at">width=</span><span class="fl">5.5</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw</span>(p_simple)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="sc">!</span><span class="fu">is.null</span>(<span class="fu">dev.list</span>()))  <span class="fu">dev.off</span>()</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="variable-power-simulation" class="section level3">
<h3>Variable power simulation</h3>
<p>Simulating a data set again with 80% of QTL effects shared across all
states and the remaining 20% of QTL effects unique to a single state.
However, in this simulation, the betas for each state are sampled using
state-specific standard deviations. This situation could occur if you
have less data available for some states and therefore expect more
variance in the estimated QTL effect sizes. Here we simulate S1-S4 with
sd=0.1 (default), S5 with sd=1, and S6 with sd=2.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>sim_sd <span class="ot">&lt;-</span> <span class="fu">qtleSimulate</span>(<span class="at">params=</span>params, <span class="at">ntests=</span>n_tests, <span class="at">nstates=</span>n_states, </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">global=</span><span class="fl">0.8</span>, <span class="at">unique=</span><span class="fl">0.2</span>, </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">beta_sd=</span><span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>sim_sd <span class="ot">&lt;-</span> <span class="fu">callSignificance</span>(sim_sd, <span class="at">assay=</span><span class="st">&quot;lfsrs&quot;</span>, </span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">thresh=</span>thresh, <span class="at">second.thresh =</span> second_thresh)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>sim_sd <span class="ot">&lt;-</span> <span class="fu">runPairwiseSharing</span>(sim_sd, <span class="at">assay=</span><span class="st">&quot;significant&quot;</span>, <span class="at">factor=</span>factor)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>p_sd <span class="ot">&lt;-</span> <span class="fu">plotPairwiseSharing</span>(sim_sd, <span class="at">col_range=</span>sharing,</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>                                <span class="at">cell_annotate =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(p_sd)</span></code></pre></div>
<p><img src="figure/02_simulation-analysis.Rmd/variable-sd-QTL-simulation-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(save){</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">paste0</span>(save.loc, <span class="st">&quot;sim_sd_pairwise.pdf&quot;</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Saving to: &quot;</span>, tmp)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(tmp, <span class="at">height=</span><span class="dv">5</span>, <span class="at">width=</span><span class="fl">5.5</span>)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw</span>(p_sd)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="sc">!</span><span class="fu">is.null</span>(<span class="fu">dev.list</span>()))  <span class="fu">dev.off</span>()</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="multi-state-simulation" class="section level3">
<h3>Multi-state simulation</h3>
<p>Simulating a dataset where states are clustered in groups, with 50%
of QTL assigned randomly to only one cluster and 50% shared across all
states. Here we simulate two clusters, but k can be set to any number
smaller than the number of states.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>sim_multi <span class="ot">&lt;-</span> <span class="fu">qtleSimulate</span>(<span class="at">params=</span>params, <span class="at">ntests=</span>n_tests, <span class="at">nstates=</span>n_states, </span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">global=</span><span class="fl">0.5</span>, <span class="at">multi=</span><span class="fl">0.5</span>, <span class="at">k=</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">2</span>))</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>sim_multi <span class="ot">&lt;-</span> <span class="fu">callSignificance</span>(sim_multi, <span class="at">assay=</span><span class="st">&quot;lfsrs&quot;</span>, </span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">thresh=</span>thresh, <span class="at">second.thresh =</span> second_thresh)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>sim_multi <span class="ot">&lt;-</span> <span class="fu">runPairwiseSharing</span>(sim_multi, <span class="at">assay=</span><span class="st">&quot;significant&quot;</span>, <span class="at">factor=</span>factor)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>p_multi1 <span class="ot">&lt;-</span> <span class="fu">plotPairwiseSharing</span>(sim_multi, <span class="at">col_range=</span>sharing,</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>                                <span class="at">cell_annotate =</span> <span class="cn">TRUE</span>,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>                                <span class="at">annotate_rows =</span> <span class="st">&quot;multistateGroup&quot;</span>)</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(p_multi1)</span></code></pre></div>
<p><img src="figure/02_simulation-analysis.Rmd/multistate-QTL-simulation-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(save){</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">paste0</span>(save.loc, <span class="st">&quot;sim_multistate_pairwise.pdf&quot;</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Saving to: &quot;</span>, tmp)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(tmp, <span class="at">height=</span><span class="dv">5</span>, <span class="at">width=</span><span class="fl">5.5</span>)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw</span>(p_multi1)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="sc">!</span><span class="fu">is.null</span>(<span class="fu">dev.list</span>()))  <span class="fu">dev.off</span>()</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>To get more detailed insight into the patterns of QTL sharing across
states, we can also visualize sharing with UpSet plots.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>p_upset <span class="ot">&lt;-</span> <span class="fu">plotUpSet</span>(sim_multi, <span class="at">annotate_by =</span> <span class="fu">c</span>(<span class="st">&quot;multistateGroup&quot;</span>, <span class="st">&quot;nSignificant&quot;</span>), </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">set_order =</span> <span class="cn">NULL</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(p_upset)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figure/02_simulation-analysis.Rmd/multi-state-upset-plot-1.png" alt="Upset plot of significant QTL tests for the multi-state simulation." width="576" />
<p class="caption">
Upset plot of significant QTL tests for the multi-state simulation.
</p>
</div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(save){</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">paste0</span>(save.loc, <span class="st">&quot;multistate_upset.pdf&quot;</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Saving to: &quot;</span>, tmp)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(tmp, <span class="at">height=</span><span class="dv">5</span>, <span class="at">width=</span><span class="dv">6</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw</span>(p_upset)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="sc">!</span><span class="fu">is.null</span>(<span class="fu">dev.list</span>()))  <span class="fu">dev.off</span>()</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
</div>
<div id="qtl-test-level-analysis" class="section level1">
<h1>QTL test level analysis</h1>
<div id="qtl-classification" class="section level2">
<h2>QTL Classification</h2>
<p>The <code>runTestMetrics</code> function outputs helpful information
about each QTL test, including the number of states where the test was
significant, the standard deviation of the betas across states, and the
classification of that QTL. Here we compare the simulate QTL types
(columns: global, multistate, unique) to the QTL classifications based
on our analysis pipeline (rows: global, multistate, unique, not
significant).</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>sim_multi <span class="ot">&lt;-</span> <span class="fu">runTestMetrics</span>(sim_multi)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>sim_stats <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">rowData</span>(sim_multi)<span class="sc">$</span>qtl_type_simple, </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">rowData</span>(sim_multi)<span class="sc">$</span>QTL)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>sim_stats[<span class="fu">c</span>(<span class="st">&quot;global&quot;</span>, <span class="st">&quot;multistate&quot;</span>, <span class="st">&quot;unique&quot;</span>), ]</span></code></pre></div>
<pre><code>            
             global multistate
  global         45          0
  multistate    181        189
  unique          1         19</code></pre>
</div>
<div id="qtl-visualizations" class="section level2">
<h2>QTL visualizations</h2>
<p>The <code>plotQTLClusters</code> function automates plotting a
heatmap of betas or any other user defined assay for tests and states.
Here we show simulated betas from the multi-state simulation and
annotate the plot with helpful information from the colData and
rowData.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(sim_multi)<span class="sc">$</span>mean_beta_abs <span class="ot">&lt;-</span> <span class="fu">abs</span>(<span class="fu">rowData</span>(sim_multi)<span class="sc">$</span>mean_beta)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>sim_multi_subset <span class="ot">&lt;-</span> <span class="fu">subset</span>(sim_multi, QTL <span class="sc">==</span> <span class="st">&quot;multistate&quot;</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>p_ms_clusters <span class="ot">&lt;-</span> <span class="fu">plotQTLClusters</span>(sim_multi_subset,</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">row_km =</span> <span class="dv">5</span>, </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">annotate_states =</span> <span class="fu">c</span>(<span class="st">&quot;multistateGroup&quot;</span>),</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">annotate_tests =</span> <span class="fu">c</span>(<span class="st">&quot;mean_beta&quot;</span>, <span class="st">&quot;nSignificant&quot;</span>))</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(p_ms_clusters)</span></code></pre></div>
<p><img src="figure/02_simulation-analysis.Rmd/plot-multistate-betas-1.png" width="576" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(save){</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">paste0</span>(save.loc, <span class="st">&quot;multistate_clusters.pdf&quot;</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Saving to: &quot;</span>, tmp)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(tmp, <span class="at">height=</span><span class="dv">7</span>, <span class="at">width=</span><span class="dv">6</span>)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw</span>(p_ms_clusters)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="sc">!</span><span class="fu">is.null</span>(<span class="fu">dev.list</span>()))  <span class="fu">dev.off</span>()</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.2.1 (2022-06-23)
 os       Rocky Linux 8.6 (Green Obsidian)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_AU.UTF-8
 ctype    en_AU.UTF-8
 tz       Australia/Melbourne
 date     2022-11-10
 pandoc   2.19.2 @ /usr/lib/rstudio-server/bin/quarto/bin/tools/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 ! package              * version  date (UTC) lib source
   abind                  1.4-5    2016-07-21 [1] CRAN (R 4.2.1)
   ashr                   2.2-54   2022-02-22 [1] CRAN (R 4.2.1)
   assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.2.1)
   backports              1.4.1    2021-12-13 [1] CRAN (R 4.2.1)
   Biobase              * 2.56.0   2022-04-26 [1] Bioconductor
   BiocGenerics         * 0.42.0   2022-04-26 [1] Bioconductor
   bit                    4.0.4    2020-08-04 [1] CRAN (R 4.2.1)
   bit64                  4.0.5    2020-08-30 [1] CRAN (R 4.2.1)
   bitops                 1.0-7    2021-04-24 [1] CRAN (R 4.2.1)
   brio                   1.1.3    2021-11-30 [1] CRAN (R 4.2.1)
   broom                  1.0.0    2022-07-01 [1] CRAN (R 4.2.1)
   bslib                  0.4.0    2022-07-16 [1] CRAN (R 4.2.1)
   cachem                 1.0.6    2021-08-19 [1] CRAN (R 4.2.1)
   Cairo                  1.6-0    2022-07-05 [1] CRAN (R 4.2.1)
   callr                  3.7.1    2022-07-13 [1] CRAN (R 4.2.1)
   car                    3.1-0    2022-06-15 [1] CRAN (R 4.2.1)
   carData                3.0-5    2022-01-06 [1] CRAN (R 4.2.1)
   checkmate              2.1.0    2022-04-21 [1] CRAN (R 4.2.1)
   circlize               0.4.15   2022-05-10 [1] CRAN (R 4.2.1)
   cli                    3.4.0    2022-09-08 [1] CRAN (R 4.2.1)
   clue                   0.3-61   2022-05-30 [1] CRAN (R 4.2.1)
   cluster                2.1.3    2022-03-28 [2] CRAN (R 4.2.1)
   codetools              0.2-18   2020-11-04 [2] CRAN (R 4.2.1)
   collapse             * 1.8.8    2022-08-15 [1] CRAN (R 4.2.1)
   colorspace             2.0-3    2022-02-21 [1] CRAN (R 4.2.1)
   ComplexHeatmap       * 2.12.1   2022-08-09 [1] Bioconductor
   cowplot              * 1.1.1    2020-12-30 [1] CRAN (R 4.2.1)
   crayon                 1.5.2    2022-09-29 [1] CRAN (R 4.2.1)
   data.table           * 1.14.4   2022-10-17 [1] CRAN (R 4.2.1)
   DBI                    1.1.3    2022-06-18 [1] CRAN (R 4.2.1)
   DelayedArray           0.22.0   2022-04-26 [1] Bioconductor
   desc                   1.4.1    2022-03-06 [1] CRAN (R 4.2.1)
   devtools               2.4.4    2022-07-20 [1] CRAN (R 4.2.1)
   digest                 0.6.30   2022-10-18 [1] CRAN (R 4.2.1)
   doParallel             1.0.17   2022-02-07 [1] CRAN (R 4.2.1)
   dplyr                * 1.0.10   2022-09-01 [1] CRAN (R 4.2.1)
   ellipsis               0.3.2    2021-04-29 [1] CRAN (R 4.2.1)
   evaluate               0.16     2022-08-09 [1] CRAN (R 4.2.1)
   fansi                  1.0.3    2022-03-24 [1] CRAN (R 4.2.1)
   fastmap                1.1.0    2021-01-25 [1] CRAN (R 4.2.1)
   fitdistrplus           1.1-8    2022-03-10 [1] CRAN (R 4.2.1)
   foreach                1.5.2    2022-02-02 [1] CRAN (R 4.2.1)
   fs                     1.5.2    2021-12-08 [1] CRAN (R 4.2.1)
   generics               0.1.3    2022-07-05 [1] CRAN (R 4.2.1)
   GenomeInfoDb         * 1.32.4   2022-09-06 [1] Bioconductor
   GenomeInfoDbData       1.2.8    2022-08-04 [1] Bioconductor
   GenomicRanges        * 1.48.0   2022-04-26 [1] Bioconductor
   GetoptLong             1.0.5    2020-12-15 [1] CRAN (R 4.2.1)
   ggplot2              * 3.3.6    2022-05-03 [1] CRAN (R 4.2.1)
   ggpubr               * 0.4.0    2020-06-27 [1] CRAN (R 4.2.1)
   ggsignif               0.6.3    2021-09-09 [1] CRAN (R 4.2.1)
   git2r                  0.30.1   2022-03-16 [1] CRAN (R 4.2.1)
   GlobalOptions          0.1.2    2020-06-10 [1] CRAN (R 4.2.1)
   glue                   1.6.2    2022-02-24 [1] CRAN (R 4.2.1)
   gridExtra              2.3      2017-09-09 [1] CRAN (R 4.2.1)
   gtable                 0.3.1    2022-09-01 [1] CRAN (R 4.2.1)
   highr                  0.9      2021-04-16 [1] CRAN (R 4.2.1)
   htmltools              0.5.3    2022-07-18 [1] CRAN (R 4.2.1)
   htmlwidgets            1.5.4    2021-09-08 [1] CRAN (R 4.2.1)
   httpuv                 1.6.5    2022-01-05 [1] CRAN (R 4.2.1)
   invgamma               1.1      2017-05-07 [1] CRAN (R 4.2.1)
   IRanges              * 2.30.1   2022-08-18 [1] Bioconductor
   irlba                  2.3.5.1  2022-10-03 [1] CRAN (R 4.2.1)
   iterators              1.0.14   2022-02-05 [1] CRAN (R 4.2.1)
   jquerylib              0.1.4    2021-04-26 [1] CRAN (R 4.2.1)
   jsonlite               1.8.3    2022-10-21 [1] CRAN (R 4.2.1)
   knitr                  1.39     2022-04-26 [1] CRAN (R 4.2.1)
   later                  1.3.0    2021-08-18 [1] CRAN (R 4.2.1)
   lattice                0.20-45  2021-09-22 [2] CRAN (R 4.2.1)
   lifecycle              1.0.2    2022-09-09 [1] CRAN (R 4.2.1)
   magick                 2.7.3    2021-08-18 [1] CRAN (R 4.2.1)
   magrittr               2.0.3    2022-03-30 [1] CRAN (R 4.2.1)
   mashr                  0.2.57   2022-01-25 [1] CRAN (R 4.2.1)
   MASS                   7.3-57   2022-04-22 [2] CRAN (R 4.2.1)
   Matrix                 1.4-1    2022-03-23 [2] CRAN (R 4.2.1)
   MatrixGenerics       * 1.8.1    2022-06-26 [1] Bioconductor
   matrixStats          * 0.62.0   2022-04-19 [1] CRAN (R 4.2.1)
   memoise                2.0.1    2021-11-26 [1] CRAN (R 4.2.1)
   mime                   0.12     2021-09-28 [1] CRAN (R 4.2.1)
   miniUI                 0.1.1.1  2018-05-18 [1] CRAN (R 4.2.1)
   mixsqp                 0.3-47   2022-10-31 [1] Github (stephenslab/mixsqp@0a80f7e)
 R multiStateQTL        * 0.1.0    &lt;NA&gt;       [?] &lt;NA&gt;
   munsell                0.5.0    2018-06-12 [1] CRAN (R 4.2.1)
   mvtnorm                1.1-3    2021-10-08 [1] CRAN (R 4.2.1)
   pillar                 1.8.1    2022-08-19 [1] CRAN (R 4.2.1)
   pkgbuild               1.3.1    2021-12-20 [1] CRAN (R 4.2.1)
   pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.2.1)
   pkgload                1.3.0    2022-06-27 [1] CRAN (R 4.2.1)
   plyr                   1.8.7    2022-03-24 [1] CRAN (R 4.2.1)
   png                    0.1-7    2013-12-03 [1] CRAN (R 4.2.1)
   prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.2.1)
   processx               3.7.0    2022-07-07 [1] CRAN (R 4.2.1)
   profvis                0.3.7    2020-11-02 [1] CRAN (R 4.2.1)
   promises               1.2.0.1  2021-02-11 [1] CRAN (R 4.2.1)
   ps                     1.7.1    2022-06-18 [1] CRAN (R 4.2.1)
   purrr                  0.3.5    2022-10-06 [1] CRAN (R 4.2.1)
   QTLExperiment        * 0.1.0    2022-11-09 [1] git2r (https://gitlab.svi.edu.au/biocellgen-public/qtlexperiment.git@42ab7e2)
   R6                     2.5.1    2021-08-19 [1] CRAN (R 4.2.1)
   RColorBrewer           1.1-3    2022-04-03 [1] CRAN (R 4.2.1)
   Rcpp                   1.0.9    2022-07-08 [1] CRAN (R 4.2.1)
   RCurl                  1.98-1.9 2022-10-03 [1] CRAN (R 4.2.1)
   remotes                2.4.2    2021-11-30 [1] CRAN (R 4.2.1)
   rjson                  0.2.21   2022-01-09 [1] CRAN (R 4.2.1)
   rlang                  1.0.5    2022-08-31 [1] CRAN (R 4.2.1)
   rmarkdown              2.15     2022-08-16 [1] CRAN (R 4.2.1)
   rmeta                  3.0      2018-03-20 [1] CRAN (R 4.2.1)
   rprojroot              2.0.3    2022-04-02 [1] CRAN (R 4.2.1)
   rstatix                0.7.0    2021-02-13 [1] CRAN (R 4.2.1)
   rstudioapi             0.14     2022-08-22 [1] CRAN (R 4.2.1)
   S4Vectors            * 0.34.0   2022-04-26 [1] Bioconductor
   sass                   0.4.2    2022-07-16 [1] CRAN (R 4.2.1)
   scales                 1.2.1    2022-08-20 [1] CRAN (R 4.2.1)
   sessioninfo            1.2.2    2021-12-06 [1] CRAN (R 4.2.1)
   shape                  1.4.6    2021-05-19 [1] CRAN (R 4.2.1)
   shiny                  1.7.2    2022-07-19 [1] CRAN (R 4.2.1)
   softImpute             1.4-1    2021-05-09 [1] CRAN (R 4.2.1)
   SQUAREM                2021.1   2021-01-13 [1] CRAN (R 4.2.1)
   stringi                1.7.8    2022-07-11 [1] CRAN (R 4.2.1)
   stringr                1.4.1    2022-08-20 [1] CRAN (R 4.2.1)
   SummarizedExperiment * 1.26.1   2022-04-29 [1] Bioconductor
   survival               3.3-1    2022-03-03 [2] CRAN (R 4.2.1)
   testthat             * 3.1.4    2022-04-26 [1] CRAN (R 4.2.1)
   tibble                 3.1.8    2022-07-22 [1] CRAN (R 4.2.1)
   tidyr                  1.2.1    2022-09-08 [1] CRAN (R 4.2.1)
   tidyselect             1.2.0    2022-10-10 [1] CRAN (R 4.2.1)
   truncnorm              1.0-8    2018-02-27 [1] CRAN (R 4.2.1)
   tzdb                   0.3.0    2022-03-28 [1] CRAN (R 4.2.1)
   urlchecker             1.0.1    2021-11-30 [1] CRAN (R 4.2.1)
   usethis                2.1.6    2022-05-25 [1] CRAN (R 4.2.1)
   utf8                   1.2.2    2021-07-24 [1] CRAN (R 4.2.1)
   vctrs                  0.4.1    2022-04-13 [1] CRAN (R 4.2.1)
   viridis                0.6.2    2021-10-13 [1] CRAN (R 4.2.1)
   viridisLite            0.4.1    2022-08-22 [1] CRAN (R 4.2.1)
   vroom                  1.6.0    2022-09-30 [1] CRAN (R 4.2.1)
   withr                  2.5.0    2022-03-03 [1] CRAN (R 4.2.1)
   workflowr              1.7.0    2021-12-21 [1] CRAN (R 4.2.1)
   xfun                   0.32     2022-08-10 [1] CRAN (R 4.2.1)
   xtable                 1.8-4    2019-04-21 [1] CRAN (R 4.2.1)
   XVector                0.36.0   2022-04-26 [1] Bioconductor
   yaml                   2.3.6    2022-10-18 [1] CRAN (R 4.2.1)
   zlibbioc               1.42.0   2022-04-26 [1] Bioconductor

 [1] /mnt/mcfiles/cazodi/R/x86_64-pc-linux-gnu-library/4.2
 [2] /opt/R/4.2.1/lib/R/library

 R ── Package was removed from disk.

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
