<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>About</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>










<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">WEEO_2021_multi-QTL-benchmark</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="01_GTEx-analysis.html">GTEx Analysis</a>
</li>
<li>
  <a href="02_simulation-analysis.html">Simulation Analysis</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://gitlab.svi.edu.au/biocellgen-private/WEEO_2021_meta-eQTL">
    <span class="fa fa-gitlab"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">About</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 1
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>QTLE_Manuscript_results/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomdunstoneaQTLEManuscriptresultstreeb8a76e33266f9a634fb449c549777c0c5901f3d0targetblankb8a76e3a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/dunstone-a/QTLE_Manuscript_results/tree/b8a76e33266f9a634fb449c549777c0c5901f3d0" target="_blank">b8a76e3</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomdunstoneaQTLEManuscriptresultstreeb8a76e33266f9a634fb449c549777c0c5901f3d0targetblankb8a76e3a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/dunstone-a/QTLE_Manuscript_results/tree/b8a76e33266f9a634fb449c549777c0c5901f3d0" target="_blank">b8a76e3</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/Adipose_Subcutaneous.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Artery_Tibial.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Esophagus_Mucosa.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Lung.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Muscle_Skeletal.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Nerve_Tibial.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Skin_Not_Sun_Exposed_Suprapubic.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Skin_Sun_Exposed_Lower_leg.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Thyroid.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    data/Whole_Blood.allpairs.txt.gz_alltests_chr1.txt
    Ignored:    output/00_Figures/
    Ignored:    output/gtex10_chr1_QTLE.rds
    Ignored:    output/gtex10_chr1_QTLE_filt.rds
    Ignored:    output/gtex_estimated_params.rds

Untracked files:
    Untracked:  QTLE_Manuscript_results.Rproj
    Untracked:  analysis/01_GTEx-analysis_old.Rmd
    Untracked:  analysis/about_old.Rmd

Unstaged changes:
    Modified:   analysis/01_GTEx-analysis.Rmd
    Deleted:    analysis/2021-07-30_pbmc8k-reference-processing.Rmd
    Deleted:    analysis/2021-08-03_pbmc8k-simulations.Rmd
    Modified:   analysis/_site.yml
    Modified:   code/preprocess_gtex_alltests.sh
    Modified:   data/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div
id="cardelino-integrating-whole-exomes-and-single-cell-transcriptomes-to-reveal-phenotypic-impact-of-somatic-variants"
class="section level2">
<h2><em>Cardelino</em> : Integrating whole exomes and single-cell
transcriptomes to reveal phenotypic impact of somatic variants</h2>
<p><strong>Key findings:</strong></p>
<ul>
<li>A new approach for integrating DNA-seq and single-cell RNA-seq data
to reconstruct clonal substructure and single-cell transcriptomes.</li>
<li>A new computational method to map single-cell RNA-seq profiles to
clones.</li>
<li>Evidence for non-neutral evolution of clonal populations in human
fibroblasts.</li>
<li>Proliferation and cell cycle pathways are commonly distorted in
mutated clonal populations, with implications for cancer and
ageing.</li>
</ul>
<p><strong>Abstract</strong></p>
<p>Decoding the clonal substructures of somatic tissues sheds light on
cell growth, development and differentiation in health, ageing and
disease. DNA-sequencing, either using bulk or using single-cell assays,
has enabled the reconstruction of clonal trees from somatic variants.
However, approaches to characterize phenotypic and functional variations
between clones are not established.</p>
<p>Here we present cardelino (<a
href="https://github.com/PMBio/cardelino"
class="uri">https://github.com/PMBio/cardelino</a>), a computational
method to assign single-cell transcriptome profiles to somatic clones
using variant information contained in single-cell RNA-seq (scRNA-seq)
data. After validating our model using simulations, we apply cardelino
to matched scRNA-seq and exome sequencing data from 32 human dermal
fibroblast lines</p>
<p>We identify hundreds of differentially expressed genes between cells
assigned to different clones. These genes were frequently enriched for
the cell cycle and pathways related to cell proliferation, and our data
point to clone gene expression phenotypes that support previous work
showing non-neutral somatic evolution in nominally healthy human skin
cells.</p>
</div>
<div id="authors" class="section level2">
<h2>Authors</h2>
<p>The full author list is as follows:</p>
<p>Davis J. McCarthy<sup>1,4,*</sup>, Raghd Rostom<sup>1,2,*</sup>,
Yuanhua Huang<sup>1,*</sup>, Daniel J. Kunz<sup>2,5,6</sup>, Petr
Danecek<sup>2</sup>, Marc Jan Bonder<sup>1</sup>, Tzachi
Hagai<sup>1,2</sup>, HipSci Consortium, Wenyi Wang<sup>8</sup>, Daniel
J. Gaffney<sup>2</sup>, Benjamin D. Simons<sup>5,6,7</sup>, Oliver
Stegle<sup>1,3,9,#</sup>, Sarah A. Teichmann<sup>1,2,5,#</sup></p>
<p><sup>1</sup>European Molecular Biology Laboratory, European
Bioinformatics Institute, Wellcome Genome Campus, CB10 1SD Hinxton,
Cambridge, UK; <sup>2</sup>Wellcome Sanger Institute, Wellcome Genome
Campus, Hinxton, CB10 1SA, UK; <sup>3</sup>European Molecular Biology
Laboratory, Genome Biology Unit, 69117 Heidelberg, Germany;
<sup>4</sup>St Vincent’s Institute of Medical Research, Fitzroy,
Victoria 3065, Australia. <sup>5</sup>Cavendish Laboratory, Department
of Physics, JJ Thomson Avenue, Cambridge, CB3 0HE, UK. <sup>6</sup>The
Wellcome Trust/Cancer Research UK Gurdon Institute, University of
Cambridge, Cambridge, CB2 1QN, UK. <sup>7</sup>The Wellcome
Trust/Medical Research Council Stem Cell Institute, University of
Cambridge, Cambridge, UK. <sup>8</sup>Department of Bioinformatics and
Computational Biology, The University of Texas MD Anderson Cancer
Center, Houston, Texas 77030, USA. <sup>9</sup>Division of Computational
Genomics and Systems Genetics, German Cancer Research Center (DKFZ),
69120, Heidelberg, Germany.</p>
<p><sup>*</sup> These authors contributed equally to this work.</p>
<p><sup>#</sup> Corresponding authors.</p>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
