---
title: 'WEEO: splatPop simulation visualization - pbmc8k'
author: "C.B. Azodi"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE)
```

```{r load-libraries-code}
suppressPackageStartupMessages({
  library(tidyverse)
  library(splatter)
  library(scater)
  library(RColorBrewer)
  library(ggpubr)
  library(cluster)
})

source("code/sce_plots.R")
save.fig <- TRUE
params.path <- "output/01_simulations/01_reference-params/pbmc8k/"
dataset <- "pbmc8k"

n.genes <- 1255
```

## 10x PBMC data

Load reference data from 10x Genomics [here](https://support.10xgenomics.com/single-cell-gene-expression/datasets/2.1.0/pbmc8k). Data contains peripheral blood mononuclear cells (PBMCs) from a healthy donor (same donor as pbmc4k). Cellranger was run with with --expect-cells=10000, and 8,381 cells were detected. Then annotate these cells using BlueprintEncodeData, with the following number of cells annotated as each cell type (note, unannotated cells are removed at this point): 

```{r load-annotate-pbmc-data, echo=FALSE}
ref <- celldex::BlueprintEncodeData()
pbmc <- TENxPBMCData(dataset = dataset)   # snapshotDate(): 2020-10-27
pbmc <- logNormCounts(pbmc)


rowData(pbmc)$Symbol <- mapIds(org.Hs.eg.db, keys=rownames(pbmc), 
                               column="SYMBOL", keytype="ENSEMBL")

pbmc <- pbmc[!is.na(rowData(pbmc)$Symbol), ]
rownames(pbmc) <- rowData(pbmc)$Symbol

pred <- SingleR(test=pbmc, ref=ref, labels=ref$label.main)

colData(pbmc) <- cbind(colData(pbmc), pred)
pbmc$pruned.labels <- gsub(" ", "_", pbmc$pruned.labels)
pbmc$pruned.labels <- as.factor(pbmc$pruned.labels)
pbmc <- pbmc[, !is.na(pbmc$pruned.labels)]

dim(pbmc)
table(pbmc$pruned.labels)
```

## Gene and cell selection

Gene wise filtering will be done to remove genes with no/little variance across cells (percent zero < 90% kept) and then the number of genes to be simulated (n=1255, # of genes on chromosome 2) will be randomly selected from the remaining genes. Filtering down our empirical data to include the same number of genes as the desired simulated dataset will ensure the library.size and gene mean estimates for splatPop reflect the real data. 

We also remove cells with fewer than 5% of genes expressed. Below is the number of genes and cells remaining after filtering. 

```{r filter-out-low-coverage-genes-and-cells}
rowData(pbmc)$percentZeros <- rowSums(counts(pbmc) == 0) / ncol(pbmc)
pbmc$percentZeros <- colSums(counts(pbmc) == 0) / nrow(pbmc)

pbmc.subset <- pbmc[rowData(pbmc)$percentZeros <= 0.9, 
                    colData(pbmc)$percentZeros <= 0.95]


set.seed(49)
pbmc.subset <- pbmc.subset[sample(1:nrow(pbmc.subset), n.genes), ]

dim(pbmc.subset)

```

The number of cells from each cell-type remaining after filtering: 

```{r cell-types-after-filtering}
table(pbmc.subset$pruned.labels)
```



