---
title: "WEEO: Multi-QTL project"
author: "Christina B Azodi"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
---

## Project overview


**Key findings:**



## Empirical Data Pre-processing

The data pre-processing for this project from the raw data described above is 
complicated and computationally expensive, so this repository does not reproduce
the data pre-processing in an automated way. However, we provide the source code
for the [Snakemake](https://snakemake.readthedocs.io/en/stable/) workflow for 
data pre-processing in this repository. Docker images providing the computing 
environment and software used are publicly available, split into an image for 
command line [bioinformatics tools](https://hub.docker.com/r/davismcc/fibroblast-clonality/)
and an [R installation](https://hub.docker.com/r/davismcc/r-singlecell-img/) with 
necessary packages installed. 

If you would like to pre-process the data from raw reads to results as we have, 
please consult our description of [how to run](data_preprocessing.html) the 
workflow. 

## Simulations

Simulated datasets were generated using [splatPop](https://www.biorxiv.org/content/10.1101/2021.06.17.448806v1) available in [Splatter v1.16+](https://www.bioconductor.org/packages/release/bioc/html/splatter.html) through Bioconductor. Realistic single-cell parameters were estimated from real data:

1. 10x PBMCs from [10x Genomics](https://support.10xgenomics.com/single-cell-gene-expression/datasets/2.1.0/pbmc8k): [Rmarkdown](2021-07-30_pbmc8k-reference-processing.html)

## Analyses

Here we present the reproducible the results of our analyses. 


The results presented in the paper were produced with these analyses.



## Data availability

This is a complicated project, and reproducing all of the results presented, especially from raw data is highly non-trivial. Nevertheless, we have made all data available so that everything is entirely reproducible.


Metadata files belong in `metadata`. Reference files belong in `references`.

With the data downloaded and organised as above, you will be able to reproduce the analyses presented in the RMarkdown files linked to above and, if desired, even run the whole analysis pipeline from raw reads to results following [these instructions](data_preprocessing.html).


-------

<a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/">Creative Commons Attribution-ShareAlike 4.0 International License</a>.

